pppd/utils.c

```diff
@@ -844,6 +926,10 @@ lock(char *dev)
 #endif
 
     while ((fd = open(lock_file, O_EXCL | O_CREAT | O_RDWR, 0644)) < 0) {
+	if (errno == ENOENT && mkdir_recursive(PPP_PATH_LOCKDIR) == 0) {
+	    /* parent dir was missing, retry now */
+	    continue;
```

Comment:
So what happens when mkdir_recursive fails? Should we at least log a message?

Reply:
It'll afll through to errno != EEXIST and print it couldn't create lock file; that's probably enough.
Looking back mkdir_recursive can leave errno set to EEXIST so we probably want to save the original ENOENT errno in that case; I'll adjust this later today.